lib/reporters/base.js

```diff
@@ -225,11 +226,21 @@ exports.list = function(failures){
  * @api public
  */
 
-function Base(runner) {
+function Base(runner, options, path) {
   var self = this
     , stats = this.stats = { suites: 0, tests: 0, passes: 0, pending: 0, failures: 0 }
     , failures = this.failures = [];
 
+  this.options = options;
+  this.path = path;
+
+  if (path) {
+    if (!fs.createWriteStream) {
```

Comment:
that gives me an idea for a server-side reporter which launches the HTML reporter and uses websockets to stream the results. :smile: